src/core/client.js

```diff
@@ -70,12 +71,30 @@ export class Client {
    */
 
   constructor(config = {}) {
-    let apiHostname = isString(config.apiHostname) ? config.apiHostname : 'https://baas.kinvey.com';
-    if (/^https?:\/\//i.test(apiHostname) === false) {
-      apiHostname = `https://${apiHostname}`;
+    let apiHostname = 'https://baas.kinvey.com';
+    let micHostname = 'https://auth.kinvey.com';
+
+    if (config.instanceId) {
+      const { instanceId } = config;
+
+      if (!isString(instanceId)) {
+        throw new KinveyError('Instance ID must be a string.');
+      }
+
+      apiHostname = `https://${instanceId}-baas.kinvey.com`;
+      micHostname = `https://${instanceId}-auth.kinvey.com`;
+    } else {
+      if (isString(config.apiHostname)) {
```

Comment:
As part of this ticket, the `micHostname` and `apiHostname` config options should be marked deprecated in the API reference. Also, the `instanceId` option should be documented in the API reference.

Reply:
I already did this. Take a look at the documentation comments above the function.

Reply:
My mistake, I see it now. Can you make the following change? 
"Deprecated: Host name used for Kinvey API requests": add the following - "Use the `instanceID` property instead"

LGTM overall.